src/routes/business/businesses.ts

```diff
@@ -0,0 +1,225 @@
+import _ from "lodash";
+import moment from "moment";
+import uuid from "node-uuid";
+
+import {
+  saveBusiness,
+  redlock,
+  setBusinessOrigin,
+  getBusiness,
+  findBusinesses,
+} from "../../db";
+
+import generateID from "../../helpers/id";
+import { storeBusinessInSortedSet } from "../../helpers/businesses";
+import { BusinessWebhookEvent, MockBusiness } from "../../helpers/types";
+import { triggerWebhook } from "../../helpers/webhooks";
+
+const businessObjectFields = [
+  "id",
+  "createdAt",
+  "name",
+  "sector",
+  "industry",
+  "industry_key",
+  "nace_code",
+  "legal_form",
+  "foundation_date",
+  "address",
+  "line_1",
+  "line_2",
+  "postal_code",
+  "city",
+  "state",
+  "country",
+  "tax_information",
+  "tax_country",
+  "tax_confirmation",
+  "registration_number",
+  "registration_issuer",
+  "fatca_relevant",
+  "fatca_crs_confirmed_at",
+  "crs_company_type",
+  "balance_sheet_total",
+  "number_employees",
+  "registration_type",
+  "registration_district",
+  "business_purpose",
+  "terms_conditions_signed_at",
+  "branch",
+  "international_operativity_expectation",
+  "vat_number",
+  "purpose_of_account_opening",
+  "source_of_funds",
+  "expected_annual_revenue",
+  "expected_annual_incoming_funds",
+  "company_status",
+];
+
+export const createBusiness = async (req, res) => {
+  const businessId = generateID(); // Do not exceed 36 characters
+  let createdBusiness;
+  const businessLockKey = `redlock:${process.env.MOCKSOLARIS_REDIS_PREFIX}:business:${businessId}`;
+
+  await redlock.using([businessLockKey], 5000, async (signal) => {
+    if (signal.aborted) {
+      throw signal.error;
+    }
+    const createdAt = moment();
+
+    const business = {
+      ...req.body,
+      id: businessId,
+      createdAt: createdAt.toISOString(),
+    };
+
+    createdBusiness = await saveBusiness(business).then(() => {
+      res.status(201).send({
+        id: businessId,
+        ...req.body,
+      });
+    });
+
+    await storeBusinessInSortedSet(business);
+
+    if (req.headers.origin) {
+      await setBusinessOrigin(businessId, req.headers.origin);
+    }
+  });
+
+  return createdBusiness;
+};
+
+export const showBusiness = async (req, res) => {
+  const { business_id: businessId } = req.params;
+  try {
+    const business = await getBusiness(businessId);
+
+    const mappedBusiness = _.pick(business, businessObjectFields);
```

Comment:
does 'pick' support nested keys? will it properly return 'address'?

Reply:
Great catch! Improved mapping added in: [f4c7f11](https://github.com/kontist/mock-solaris/pull/202/commits/f4c7f11bcfb7334a24d7a0432fe1d901c7440497)

For reference: [Can _.pick() be used with nested objects?](https://www.google.com/search?q=lodash+pick+support+nested+object&oq=lodash+pick+support+nested&gs_lcrp=EgZjaHJvbWUqBwgBECEYoAEyBggAEEUYOTIHCAEQIRigATIHCAIQIRigATIHCAMQIRifBTIHCAQQIRifBTIHCAUQIRifBdIBCDY1NDZqMGo3qAIAsAIA&sourceid=chrome&ie=UTF-8#:~:text=Lodash%20_.pick,_%2Dpick%2Dmethod)